network/src/noise/error.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use libra_crypto::noise::NoiseError;
+use libra_types::PeerId;
+use short_hex_str::ShortHexStr;
+use std::io;
+use thiserror::Error;
+
+/// Different errors than can be raised when negotiating a Noise handshake.
+#[derive(Debug, Error)]
+pub enum NoiseHandshakeError {
+    #[error("noise client: missing remote server's public key when dialing")]
+    MissingRemotePublicKey,
+
+    #[error("noise client: error building handshake init message: {0}")]
+    BuildClientHandshakeMessage(NoiseError),
+
+    #[error("noise client: error sending client handshake init message: {0}")]
+    ClientWrite(io::Error),
+
+    #[error(
+        "noise client: error reading server handshake response message, server \
+         probably rejected our handshake message: {0}"
+    )]
+    ClientRead(io::Error),
+
+    #[error("noise client: error finalizing secure connection: {0}")]
+    ClientFinalize(NoiseError),
+
+    #[error("noise server: error reading client handshake init message: {0}")]
+    ServerRead(io::Error),
+
+    #[error("noise server: client peer id is malformed somehow: {0}")]
+    InvalidClientPeerId(String),
+
+    #[error("noise server: detected self-dial: we're trying to connect to ourselves")]
+    SelfDial,
+
+    #[error("noise server: client {0}: client is expecting us to have the wrong public key: {1}")]
+    ClientExpectingWrongPubkey(ShortHexStr, String),
+
+    #[error("noise server: client {0}: error parsing handshake init message: {1}")]
+    ServerParseClient(ShortHexStr, NoiseError),
+
+    #[error("noise server: client {0}: lock is poisoned")]
+    LockPoisoned(ShortHexStr),
+
+    #[error(
+        "noise server: client {0}: known client connecting to us with \
+         unauthenticated public key: {1}"
+    )]
+    UnauthenticatedClientPubkey(ShortHexStr, String),
+
+    #[error("noise server: client {0}: client connecting with unauthenticated peer id: {1}")]
+    UnauthenticatedClient(ShortHexStr, PeerId),
```

Comment:
"unauthenticated peer id" pretty much means "not a validator peer id" in the validator network context

Reply:
Maybe I wasn't being clear, but I'm saying if I'm some operator, and I see `"unauthenticated peer id"`.  I would be thinking, does this mean the peer id has to be authenticated with some key etc.  Rather than the client is connecting with an `unrecognized`, or `unexpected` peer id.

Similar to the `"unauthenticated public key"`, how do I authenticate a `public key`, when it's a recognized peer.  It makes more sense to be a `unrecognized` or `unexpected` public key.  Especially, since the authentication would be done with that public key.

Reply:
would be good to clarify here why this is an error... such that the network is expecting only authenticated connections or that the key is unknown for this peer id